### PR TITLE
chore: user agent updates

### PR DESCRIPTION
**Pull request by Shahroz16**
closes: https://linear.app/customerio/issue/MBL-106/user-agent

- Adds user-agent in payload
- Unify plugin to update context properties changes
- Added test

### CODE REVIEW

File: Sources/Common/Service/UserAgentUtil.swift [hunk 1]
```diff
@@ -4,14 +4,12 @@ public protocol UserAgentUtil: AutoMockable {
     func getUserAgentHeaderValue() -> String
 }
 
-// sourcery: InjectRegister = "UserAgentUtil"
+// sourcery: InjectRegisterShared = "UserAgentUtil"
 public class UserAgentUtilImpl: UserAgentUtil {
     private let deviceInfo: DeviceInfo
-    private let sdkWrapperConfig: SdkWrapperConfig?
```
Comment (mrehan27) [inline Sources/Common/Service/UserAgentUtil.swift:10]: If `sdkWrapperConfig` is not being used now, can we remove it completely? I'm okay if we do it in another PR or ticket, but just wanted to check if that simply needs to be removed, we can probably do that.

File: Sources/DataPipeline/Plugins/Context.swift [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
     public let type = PluginType.before
     public weak var analytics: Analytics?
 
-    public var token: String?
+    public var deviceToken: String?
     public var attributes: [String: Any] = [:]
 
+    var userAgentUtil: UserAgentUtil {
+        DIGraphShared.shared.userAgentUtil
+    }
```
Comment (mrehan27) [inline Sources/DataPipeline/Plugins/Context.swift:14]: I would prefer it as `let` and initializing in constructor so it is assigned only once 🤓

**Review by ami-aman — APPROVED** [review]

**Review by mrehan27 — APPROVED** [review]
Looks good, just a few comments 👍🏻

File: Sources/Common/Service/UserAgentUtil.swift (hunk 1 above)
Comment (Shahroz16) [reply Sources/Common/Service/UserAgentUtil.swift:10]: i wanted to do it a separate PR but just doing it in this PR might be easier 😆
